### PR TITLE
chore: better request handling logging

### PR DESCRIPTION
**Pull request by dpc**
Mostly about adding an ID, to help filter interleaved logging messages. But also I've noticed there was another span already there, just one that was not using `LOG_` constants.

### CODE REVIEW

File: fedimint-core/src/module/mod.rs [hunk 1]
```diff
@@ -364,13 +367,6 @@ impl ApiEndpoint<()> {
         E::Param: Debug,
         E::Response: Debug,
     {
-        #[instrument(
-            target = "fedimint_server::request",
-            level = "trace",
-            skip_all,
-            fields(method = E::PATH),
-            ret,
-        )]
```
Comment (maan2003) [inline fedimint-core/src/module/mod.rs:373]: you can do `fields(id = REQ_ID.fetch_add(1, Ordering::SeqCst))` btw

Comment (maan2003) [reply fedimint-core/src/module/mod.rs:373]: or is it instrument not supporting CONST target

File: fedimint-core/src/module/mod.rs [hunk 2]
```diff
@@ -356,6 +356,9 @@ pub struct ApiEndpoint<M> {
     pub handler: HandlerFn<M>,
 }
 
+/// Global request ID used for logging
+pub static REQ_ID: AtomicU64 = AtomicU64::new(0);
```
Comment (maan2003) [inline fedimint-core/src/module/mod.rs:360]: why pub

File: fedimint-core/src/module/mod.rs (hunk 1 above)
Comment (dpc) [reply fedimint-core/src/module/mod.rs:373]: It was actually the `ID`, thanks, but while at it, I can also use const target, so let's just keep it this way.

Comment (maan2003) [reply fedimint-core/src/module/mod.rs:373]: you can even do like 

```
fields(id = {
  static REQ_ID: AtomicU64 = AtomicU64::new(0); 
  REQ_ID.fetch_add(1, Ordering::SeqCst)) 
})

File: fedimint-core/src/module/mod.rs (hunk 2 above)
Comment (dpc) [reply fedimint-core/src/module/mod.rs]: Just general sloppiness.

**Review by maan2003 — APPROVED** [review]

File: fedimint-core/src/module/mod.rs [hunk 3]
```diff
@@ -400,7 +396,15 @@ impl ApiEndpoint<()> {
                         .to_typed()
                         .map_err(|e| ApiError::bad_request(e.to_string()))?;
 
-                    let ret = handle_request::<E>(m, &mut context, request).await?;
+                    let span = tracing::info_span!(
+                        target: LOG_NET_API,
+                        "api_req",
+                        id = REQ_ID.fetch_add(1, Ordering::SeqCst),
```
Comment (elsirion) [inline fedimint-core/src/module/mod.rs:402]: Does JSON RPC allow sending some connection identifier? Would be great if we could track these requests across the gateway/client <> federation boundary (I think that's how distributed tracing is supposed to work?).

Comment (dpc) [reply fedimint-core/src/module/mod.rs:402]: The client picks its own request ID and server responds with it.

But I'm not aware of any place to put a server-side ID in the response.

**Pull request merged**